### PR TITLE
Pulsar topic terraform provider

### PR DESCRIPTION
**Pull request by wangjia007bond**
- Pulsar topic terraform provider
- Small refactor and fix spelling

### CODE REVIEW

**Comment by sijie** [comment]
@wolfstudy can you review this pull request?

File: pulsar/resource_pulsar_topic.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+func resourcePulsarTopicExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	_, find, err := getTopic(d, meta)
+	if err != nil {
+		return false, fmt.Errorf("ERROR_READ_TOPIC: %w", err)
+	}
+
+	return *find, nil
+}
+
+func getTopic(d *schema.ResourceData, meta interface{}) (*utils.TopicName, *bool, error) {
```
[the remark this answers is not in the document]
Comment (wangjia007bond) [reply pulsar/resource_pulsar_topic.go]: The reason I want to use pointer is I want to make explicit clear that when error occur, that not means we are not found topic, by eliminate someone ignore the error but use found bool to indicate the topic found or not.

File: pulsar/resource_pulsar_topic.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+	for _, topic := range append(partitionedTopics, nonPartitionedTopics...) {
+		if topicName.String() == topic {
+			return topicName, &found, nil
+		}
+	}
+
+	return nil, &notFound, nil
+}
+
+func unmarshalTopicNameAndPartitions(d *schema.ResourceData) (*utils.TopicName, *int, error) {
```
[the remark this answers is not in the document]
Comment (wangjia007bond) [reply pulsar/resource_pulsar_topic.go]: Same reason I want to use partitions just indicate real partition number, using error to indicate any error occurs. another way is if error occur, put -1 to partitions, but that just apply another meaning to partitions.

[the remark this answers is not in the document]
Comment (wolfstudy) [reply pulsar/resource_pulsar_topic.go]: @wangjia007bond Sorry missing the reply.

It is correct to use pointer and value here. 

In `getTopic()`, we define two variables of bool, if we use point, we need to convert between value and pointer, but in fact, we do not need to do these operations, that's why I recommend using `bool` instead `*bool`.

> Same reason I want to use partitions just indicate real partition number, using error to indicate any error occurs. another way is if error occur, put -1 to partitions, but that just apply another meaning to partitions.

I agree with you, we can use `found` and `notFound`,

[the remark this answers is not in the document]
Comment (wolfstudy) [reply pulsar/resource_pulsar_topic.go]: For example:

in `getTopic()`, we return the `*bool`, but the `found` is a `value` of bool, so when return the value, we need to do as follows:

```
return nil, &notFound, nil
```

In `resourcePulsarTopicRead()`, the `find` is a `value` of bool, so we also need to do as follows:

```
topicName, find, err := getTopic(d, meta)
	if !*find && err != nil {
		d.SetId("")
		return nil
	}
```

[the remark this answers is not in the document]
Comment (wangjia007bond) [reply pulsar/resource_pulsar_topic.go]: change from *bool to bool.

File: pulsar/resource_pulsar_topic.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (wangjia007bond) [reply pulsar/resource_pulsar_topic.go]: change from *bool to bool.

**Comment by wolfstudy** [comment]
@sijie The change LGTM, please take a look again.

**Pull request merged**